net/node/infoUpdate.go

```diff
@@ -104,10 +104,30 @@ func (node *node) ReqNeighborList() {
 }
 
 func (node *node) ConnectSeeds() {
-	if node.nbrNodes.GetConnectionCnt() == 0 {
+	if node.nbrNodes.GetConnectionCnt() < MINCONNCNT {
 		seedNodes := config.Parameters.SeedList
 		for _, nodeAddr := range seedNodes {
-			go node.Connect(nodeAddr)
+			established := false
+			var n Noder
+			var ip net.IP
+			node.nbrNodes.Lock()
+			for _, tn := range node.nbrNodes.List {
+				addr := getNodeAddr(tn)
+				ip = addr.IpAddr[:]
+				addrstring := ip.To16().String() + ":" + strconv.Itoa(int(addr.Port))
+				if tn.GetState() == ESTABLISH && nodeAddr == addrstring {
```

Comment:
tks for reviewing
> the connecting state like HANDSHAKE, HAND, INIT should also be considered
This fix mainly focus on the issue that consensus node cannot connect enough nodes in some scenarios, ReqNeighborList can fetch the newest neighbor list of seeds, if the state in not ESTABLISH,  the connection is not stable and I am not sure that the request can be delivered correctly. If state changed to ESTABLISH in the end, ReqNeighborList would be called.

> if SeedList is [node1], and nbrNodes.List is [node2, node3] , it will not call ReqNeighborList function at all

Yes, that means we can not connect to the seed node, the fix try to ask the seed for newest  node address. if the connection is lost we try to recover so the consensus can running again,  if the node is not a consensus node, it is ok if node only have 2 neighbor nodes OR we can create another fix for the nvp node.